Countries InfoTests/Tests/CountriesListViewControllerTests.swift

```diff
@@ -0,0 +1,240 @@
+        XCTAssertEqual(sut.countriesCollectionView.delegate as? CountriesListViewController, sut.countriesCollectionView.dataSource as? CountriesListViewController)
+    }
+    
+    func testSutConformsToCollectionViewDataSource() {
+        XCTAssert(sut.conforms(to: UICollectionViewDataSource.self))
+        XCTAssertTrue(sut.responds(to: (#selector(sut.collectionView(_:numberOfItemsInSection:)))))
+        XCTAssertTrue(sut.responds(to: (#selector(sut.collectionView(_:cellForItemAt:)))))
+    }
+    
+    func testSutConformsToCollectionViewDelegate() {
+        
+        XCTAssert(sut.conforms(to: UICollectionViewDelegate.self))
+        XCTAssertTrue(sut.responds(to: #selector(sut.collectionView(_:didSelectItemAt:))))
+    }
+    
+    func testSutConformsToCollectionViewDelegateFlowLayout () {
+        
+        XCTAssert(sut.conforms(to: UICollectionViewDelegateFlowLayout.self))
+        XCTAssertTrue(sut.responds(to: #selector(sut.collectionView(_:layout:sizeForItemAt:))))
+        XCTAssertTrue(sut.responds(to:#selector(sut.collectionView(_:layout:insetForSectionAt:))))
+        XCTAssertTrue(sut.responds(to:#selector(sut.collectionView(_:layout:minimumLineSpacingForSectionAt:))))
+        XCTAssertTrue(sut.responds(to:#selector(sut.collectionView(_:layout:minimumInteritemSpacingForSectionAt:))))
+    }
+    
+    func testNumberOfSection() {
+        let mockApolloManager = ApolloManagerMock()
+        sut.viewModel  = CountriesListViewModel(apollo: mockApolloManager)
+        
+        let numberOfsection = sut.countriesCollectionView.numberOfSections
+        XCTAssertEqual(numberOfsection, 1)
+    }
+    
+    func testNumberOfItem() {
+        let mockApolloManager = ApolloManagerMock()
+        sut.viewModel  = CountriesListViewModel(apollo: mockApolloManager)
+        
+        sut.viewModel.countries.append(Country(name: "Foo", capital: "Bar", population: 10, currencies: [], officialLanguages: [], timezones: [], callingCodes: [], alpha2Code: "uk", subregion: nil))
+        XCTAssertEqual(sut.countriesCollectionView.numberOfItems(inSection: 0), 2)
+        sut.viewModel.countries.append(Country(name: "Foo", capital: "Bar", population: 10, currencies: [], officialLanguages: [], timezones: [], callingCodes: [], alpha2Code: "uk", subregion: nil))
+        sut.countriesCollectionView.reloadData()
+        XCTAssertEqual(sut.countriesCollectionView.numberOfItems(inSection: 0), 3)
+    }
+    
+    func testCellForItem() {
+        let mockApolloManager = ApolloManagerMock()
+        sut.viewModel  = CountriesListViewModel(apollo: mockApolloManager)
+        
+        let indexPath = IndexPath(row: 0, section: 0)
+        let cell = sut.countriesCollectionView.dataSource?.collectionView(sut.countriesCollectionView, cellForItemAt: indexPath) as? CountryViewCell
+        XCTAssertEqual(cell?.countryNameLabel.text , sut.viewModel.countries.first?.name)
+    }
+    
+    func testApolloManagerNetworkFetch() {
```

Comment:
It looks these test should live in separate file, since they are specific to Apollo